backend/clickhouse/log_row.go

```diff
@@ -107,50 +110,37 @@ func WithProjectIDString(projectID string) LogRowOption {
 	}
 }
 
-func cast[T string | int64 | float64](v interface{}, fallback T) T {
-	c, ok := v.(T)
-	if !ok {
-		return fallback
-	}
-	return c
-}
-
-func GetAttributesMap(resourceAttributes, spanAttributes, eventAttributes map[string]any, isFrontendLog bool) map[string]string {
+func GetAttributesMap(ctx context.Context, resourceAttributes, spanAttributes, eventAttributes map[string]any, isFrontendLog bool) map[string]string {
 	attributesMap := make(map[string]string)
 	for _, m := range []map[string]any{resourceAttributes, spanAttributes, eventAttributes} {
 		for k, v := range m {
-			shouldSkip := false
+			prefixes := highlight.InternalAttributePrefixes
+			if isFrontendLog {
+				prefixes = append(prefixes, highlight.BackendOnlyAttributePrefixes...)
+			}
 
-			for _, attr := range highlight.InternalAttributes {
-				if k == attr {
+			shouldSkip := false
+			for _, prefix := range prefixes {
+				if strings.HasPrefix(k, prefix) {
 					shouldSkip = true
 					break
 				}
 			}
-
-			if isFrontendLog {
-				for _, attr := range highlight.BackendOnlyAttributePrefixes {
-					if strings.HasPrefix(k, attr) {
-						shouldSkip = true
-						break
-					}
-				}
-			}
-
 			if shouldSkip {
 				continue
 			}
 
-			vStr := cast(v, "")
-			if vStr != "" {
+			if vStr, ok := v.(string); ok {
+				if len(vStr) > LogAttributeValueLengthLimit {
+					log.WithContext(ctx).Warnf("attribute value for %s is too long %d", k, len(vStr))
```

Comment:
It might be helpful to get an idea of which of `resourceAttributes, spanAttributes, eventAttributes` actually passed in a large value. As it is now, it would be really challenging to reverse engineer the offending source.

Reply:
improved the log here